File: Module 2/25_IndirectRecursion/Python/IndirectRecursion.py
def func2(n):
    if n <= 5:
        print(n, end=" ")
        func1(n + 1)

def func1(n):
    if n <= 5:
        print(n, end=" ")
        func2(n + 1)

if __name__ == "__main__":
    func1(1)

File: Module 2/25_IndirectRecursion/C/IndirectRecursion.c
#include <stdio.h>

void func1(int n);
void func2(int n);

void func2(int n) {
    if (n <= 5) {
        printf("%d ", n);
        func1(n+1);
    }
}

void func1(int n) {
    if (n <= 5) {
        printf("%d ", n);
        func2(n+1);
    }
}

void main() {
    func1(1);
}